7. Answer the following questions  using dataset provided below

    1. What is the % of sessions with an intention to book a scooter?
    2. What is the % of sessions with a successful booking? 
    3. What determines, according to you, an intention to book?
    4. What are factors, according to you, determining a successful booking? 
    5. Do you have any hypotheses after exploring this data? If yes, please add them.  

https://drive.google.com/file/d/1MM1vzR0nuAfFH8tIXzs7PJMulZ9Nod22/view?usp=sharing

In [1]:
#Import the libraries
import pandas as pd
import numpy as np
import requests
import matplotlib.pyplot as plt
import io

In [4]:
#Download and load the dataset
file_id = '1MM1vzR0nuAfFH8tIXzs7PJMulZ9Nod22'
url = f'https://drive.google.com/uc?id={file_id}'
download = requests.get(url).content
# Read the CSV file into a Pandas DataFrame
df = pd.read_csv(io.StringIO(download.decode('utf-8')),sep = ";")

In [5]:
#View the data
df.head(10)

anonymous_id business_zoned context_app_version  \
0  9DDC5BCD-0477-4E83-A32A-51013DA0A4E7        Hamburg              3.1.13   
1  5592E78E-CE87-4258-B8D7-DD834A687A1E        Hamburg              3.1.13   
2  5592E78E-CE87-4258-B8D7-DD834A687A1E        Hamburg              3.1.13   
3  5592E78E-CE87-4258-B8D7-DD834A687A1E        Hamburg              3.1.13   
4  5592E78E-CE87-4258-B8D7-DD834A687A1E        Hamburg              3.1.13   
5  5592E78E-CE87-4258-B8D7-DD834A687A1E        Hamburg              3.1.13   
6  9DDC5BCD-0477-4E83-A32A-51013DA0A4E7        Hamburg              3.1.13   
7  9DDC5BCD-0477-4E83-A32A-51013DA0A4E7        Hamburg              3.1.13   
8  9DDC5BCD-0477-4E83-A32A-51013DA0A4E7        Hamburg              3.1.13   
9  9DDC5BCD-0477-4E83-A32A-51013DA0A4E7        Hamburg              3.1.13   

                      context_device_id context_os_name  \
0  7A718FF6-584F-40E2-939D-224231295B1E             iOS   
1  11D0CB16-DA38-4565-9C3B-066E7877856C             iOS   
2  11D0CB16-DA38-4565-9C3B-066E7877856C             iOS   
3  11D0CB16-DA38-4565-9C3B-066E7877856C             iOS   
4  11D0CB16-DA38-4565-9C3B-066E7877856C             iOS   
5  11D0CB16-DA38-4565-9C3B-066E7877856C             iOS   
6  7A718FF6-584F-40E2-939D-224231295B1E             iOS   
7  7A718FF6-584F-40E2-939D-224231295B1E             iOS   
8  7A718FF6-584F-40E2-939D-224231295B1E             iOS   
9  7A718FF6-584F-40E2-939D-224231295B1E             iOS   

                                          event_name          received_at  \
0                    Out-Of-Business Zone View Shown  2020-09-01 23:59:44   
1                                 Application Opened  2020-09-01 23:59:41   
2                             Ride Done - Successful  2020-09-01 23:59:41   
3                                 Application Opened  2020-09-01 23:59:36   
4                           Ride Done - Unsuccessful  2020-09-01 23:59:36   
5                                 Application Opened  2020-09-01 23:59:36   
6                                 Application Opened  2020-09-01 23:59:14   
7  Reduced Speed Snackbar Shown - Out Of Business...  2020-09-01 23:59:14   
8                    Out-Of-Business Zone View Shown  2020-09-01 23:59:14   
9                                 Application Opened  2020-09-01 23:58:58   

            created_at  
0  2020-09-01 23:59:18  
1  2020-09-01 23:59:35  
2  2020-09-01 23:59:37  
3  2020-09-01 23:37:46  
4  2020-09-01 23:37:52  
5  2020-09-01 23:57:45  
6  2020-09-01 23:58:59  
7  2020-09-01 23:59:03  
8  2020-09-01 23:59:03  
9  2020-09-01 23:58:58

In [6]:
#View the shape of the dataframe
df.shape

(102422, 8)

In [7]:
#find the unique values of Event_name
df['event_name'].unique()

array(['Out-Of-Business Zone View Shown', 'Application Opened',
       'Ride Done - Successful', 'Ride Done - Unsuccessful',
       'Reduced Speed Snackbar Shown - Out Of Business Zone',
       'Vehicle Selected - User', 'Vehicle Selected - Auto',
       'Qr Scanner Tapped', 'Qr Scanner - Scanned Succeeded',
       'Ride Started - Pre-Ride', 'Ride Started - Successful',
       'Application Updated', 'Scooter Selected - User',
       'Scooter Selected - Auto',
       'Qr Scanner - Scan Failed - Vehicle Not Available',
       'Paypal Added Successfully', 'List Of Payment Shown',
       'Payment Method Chosen', 'Personal Details Added Succeeded',
       'Payment Method Screen Opened', 'Phone Number Entered',
       'Terms And Privacy Policy Accepted', 'Qr Scanner - Enter Number',
       'Ride Started - Unsuccessful', 'Add Payment Method',
       'Sidebar Opened', 'Redeem Code', 'Play Sound From Vehicle',
       'Map Legend Clicked', 'Rating Screen Closed - Maybe Later',
       'Payment De

In [8]:
sorted(df['event_name'].unique())

['Account Setup Opened',
 'Add Payment Method',
 'Application Opened',
 'Application Updated',
 'Credit Card Details Accepted',
 'Credit Card Details Rejected',
 'Customer Blocked - Phone Validation',
 'Help - Booking',
 'Help - Sidebar',
 'Layover - Tier Not Available Yet',
 'List Of Payment Shown',
 'Location Services Dialogue Opened',
 'Location Services Enabled',
 'Map Legend Clicked',
 'No-Parking View Shown',
 'Out-Of-Business Zone View Shown',
 'Payment Details Screen Opened',
 'Payment Method Chosen',
 'Payment Method Screen Opened',
 'Payment Opened',
 'Paypal Added Successfully',
 'Paypal Cancelled',
 'Paypal Dialogue - Cancel Pressed',
 'Paypal Dialogue - Okay Pressed',
 'Paypal Dialogue Shown',
 'Paypal Error Shown',
 'Personal Details Added Succeeded',
 'Phone Number Blocked - Phone Validation',
 'Phone Number Entered',
 'Play Sound From Scooter',
 'Play Sound From Vehicle',
 'Qr Scanner - Cancel',
 'Qr Scanner - Enter Number',
 'Qr Scanner - Flashlight',
 'Qr Scanner - Sc

In [9]:
#Check the Null values
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 102422 entries, 0 to 102421
Data columns (total 8 columns):
 #   Column               Non-Null Count   Dtype 
---  ------               --------------   ----- 
 0   anonymous_id         102422 non-null  object
 1   business_zoned       102422 non-null  object
 2   context_app_version  102422 non-null  object
 3   context_device_id    102422 non-null  object
 4   context_os_name      102422 non-null  object
 5   event_name           102422 non-null  object
 6   received_at          102422 non-null  object
 7   created_at           102422 non-null  object
dtypes: object(8)
memory usage: 6.3+ MB


There are no null values in the dataset

What is the % of sessions with an intention to book a scooter?

In [14]:
(df[(df['event_name'] == 'Scooter Selected - Auto') | (df['event_name'] == 'Scooter Selected - User')]['event_name'].count()/len(df))*100

15.36193395950089

15.36 % session having intention to buy scooter

What is the % of sessions with a successful booking?

In [17]:
(df[df['event_name'].isin(['Ride Started - Successful','Play Sound From Scooter','Ride Done - Successful','Play Sound From Vehicle'])]['event_name'].count()/len(df))*100

10.620765070004492

10.6% of sessions with a successful booking

What determines, according to you, an intention to book?

Events like below have intention to book.

 1. 'Scooter Selected - Auto',
 2. 'Scooter Selected - User',
 3.  'Vehicle Selected - Auto',
 4. 'Vehicle Selected - User'
 5.  'Payment Details Screen Opened',
 6. 'Payment Method Chosen',
 7. 'Payment Method Screen Opened',
 8. 'Payment Opened',
 9. 'Paypal Added Successfully',

What are factors, according to you, determining a successful booking?

A successful booking typically involves a series of events such as selecting a scooter, entering payment information, and starting a ride. These events should occur in a sequence that indicates a user's intention to book a scooter and their successful completion of that booking.

Do you have any hypotheses after exploring this data? If yes, please add them.

After exploring the data, it's clear that this dataset contains various events related to a vehicle rental service. However, the dataset lacks explicit labels or context for events related to booking and successful bookings. Therefore, it's challenging to form hypotheses about user behavior or intentions based solely on this dataset.